src/test/java/jewellery/inventory/helper/ProductTestHelper.java

```diff
@@ -0,0 +1,66 @@
+package jewellery.inventory.helper;
+
+import jewellery.inventory.dto.request.ProductRequestDto;
+import jewellery.inventory.dto.request.resource.ResourceQuantityRequestDto;
+import jewellery.inventory.model.Product;
+import jewellery.inventory.model.ResourceInUser;
+import jewellery.inventory.model.User;
+import jewellery.inventory.model.resource.Resource;
+import jewellery.inventory.model.resource.ResourceInProduct;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+import java.util.UUID;
+
+public class ProductTestHelper {
+
+    public static ResourceInUser createResourceInUser(User user, Resource pearl) {
+        ResourceInUser resourceInUser = new ResourceInUser();
+        resourceInUser.setId(UUID.randomUUID());
+        resourceInUser.setOwner(user);
+        resourceInUser.setResource(pearl);
+        resourceInUser.setQuantity(20);
+        return resourceInUser;
+    }
+
+    public static ProductRequestDto createProductRequestDto(User user, ResourceQuantityRequestDto resourceQuantityRequestDto) {
+        ProductRequestDto productRequestDto = new ProductRequestDto();
+        productRequestDto.setOwnerId(user.getId());
+        productRequestDto.setName("TestProductDto");
```

Comment:
I believe the productRequestDto and the Product entity should hold the same properties for most of the test cases. Otherwise you wouldn't be able to compare them properly afterwards. Consider extracting common values like name, description, salePrice to constants.